pkg/workloads/console/webhook.go

```diff
@@ -63,3 +93,34 @@ func (c *ConsoleAuthenticator) Handle(ctx context.Context, req types.Request) ty
 
 	return admission.PatchResponse(csl, copy)
 }
+
+type ConsoleTemplateExistenceValidator ConsoleWebhookHandler
+
+func (c *ConsoleTemplateExistenceValidator) Handle(ctx context.Context, req types.Request) types.Response {
+	logger := kitlog.With(c.logger, "uuid", string(req.AdmissionRequest.UID))
+	logger.Log("event", "request.start")
+	defer func(start time.Time) {
+		logger.Log("event", "request.end", "duration", time.Now().Sub(start).Seconds())
+	}(time.Now())
+
+	csl := &workloadsv1alpha1.Console{}
+	if err := c.decoder.Decode(req, csl); err != nil {
+		admission.ErrorResponse(http.StatusBadRequest, err)
+	}
+
+	// validate that the console template ref points to a real console template
+	consoleTemplateName := kubeTypes.NamespacedName{
```

Comment:
(As discussed IRL) I think we're going to want to remove this validation, as there's no guarantee when you upload your resource that you're not about to upload your console template, and neither helm nor any kube tooling make guarantees about resource creation ordering.

I think validating this in the controller is better: we're going to have to do this anyway, as just because the template is there when we create it doesn't mean it won't be deleted some other time, and it avoids a coupling between the two resources. We lose the immediacy of feedback but that's more of a Kubernetes API design philosophy rather than something we get to decide.